cache/manager.go

```diff
@@ -1009,6 +1026,9 @@ func (cm *cacheManager) createDiffRef(ctx context.Context, parents parentRefs, d
 }
 
 func (cm *cacheManager) Prune(ctx context.Context, ch chan client.UsageInfo, opts ...client.PruneInfo) error {
+	ctx, span := cm.tracer.Start(ctx, "(*cacheManager).Prune")
```

Comment:
The prunes coming from background GC should have a root span based on the GC function, not for individual prune calls for each rule. (Can even trace the reason for starting the GC).

For Prunes coming through grpc they should trace to the tracing context already in `ctx`.

So afaics these should use the `tracing.StartSpan(ctx)` like the rest of the tracers and `control.gc()` should have a way to inject root span to `ctx` before it calls into `cm.Prune()`.